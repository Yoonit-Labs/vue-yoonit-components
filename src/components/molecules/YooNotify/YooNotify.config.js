module.exports = {
  notifyFill: {
    options: [
      'primary',
      'danger',
      'light',
      'dark'
    ]
  },
  notifyTextColor: {
    options: [
      'light',
      'dark'
    ]
  },
  notifyTextSize: {
    options: [
      'xxxs',
      'xxs',
      'xs',
      'sm',
      'md',
      'lg',
      'xl',
      'xxl',
      'xxxl',
      'display'
    ]
  }
}
