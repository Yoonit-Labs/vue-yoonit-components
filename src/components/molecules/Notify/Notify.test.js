import { mount, shallowMount } from '@vue/test-utils'
import YooNotify from '@/components/molecules/Notify/Notify.vue'
import PropsConfig from './Notify.config'

const classBlock = 'yoo-notify'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return mount(YooNotify, {
    slots: { default: SlotText }
  })
}

describe('YooNotify Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', async () => {
    expect(wrapper.classes(`${classBlock}`)).toBe(true)
    expect(wrapper.find('.yoo-notify__text').exists()).toBe(true)
    expect(wrapper.find('.yoo-indicator').exists()).toBe(true)
    expect(wrapper.find('.yoo-btn').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('showIndicator', () => {
      it('Has a valid default value', () => {
        expect(YooNotify.props.showIndicator.default).toBe(true)
      })
      it('Test prop showIndicator', async () => {
        await wrapper.setProps({ showIndicator: false })
        expect(wrapper.find(`.${classBlock}`).attributes().style).toContain('grid-template-columns: auto;')
      })
    })
    describe('showIcon', () => {
      it('Has a valid default value', () => {
        expect(YooNotify.props.showIcon.default).toBe(false)
      })
      it('Test prop showIndicator', async () => {
        await wrapper.setProps({ showIndicator: false, showIcon: true, showButton: true })
        expect(wrapper.find(`.${classBlock}__icon`).exists()).toBe(true)
      })
    })
    describe('iconStyle', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.iconStyle.options.includes(YooNotify.props.iconStyle.default)).toBe(true)
      })
      PropsConfig.iconStyle.options.forEach(iconStyle => {
        it(`Includes iconStyle class: .${iconStyle === 'solid' ? 'fas' : 'far'}`, async () => {
          await wrapper.setProps({ showIndicator: false, showIcon: true, icon: 'cog', iconStyle })
          expect(wrapper.find('.yoo-notify__icon').classes().includes(`${iconStyle === 'solid' ? 'fas' : 'far'}`)).toBe(true)
        })
      })
    })
    describe('notifyFill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyFill.options.includes(YooNotify.props.notifyFill.default)).toBe(true)
      })
      PropsConfig.notifyFill.options.forEach(notifyFill => {
        it(`Includes notifyFill class: .${classBlock}--${notifyFill}`, async () => {
          await wrapper.setProps({ notifyFill })
          expect(wrapper.find(`.${classBlock}--${notifyFill}`).exists()).toBe(true)
        })
      })
    })
    describe('notifyTextColor', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyTextColor.options.includes(YooNotify.props.notifyTextColor.default)).toBe(true)
      })
      PropsConfig.notifyTextColor.options.forEach(notifyTextColor => {
        it(`Includes notifyTextColor class: .${classBlock}__text--${notifyTextColor}`, async () => {
          await wrapper.setProps({ notifyTextColor })
          expect(wrapper.find(`.${classBlock}__text--${notifyTextColor}`).exists()).toBe(true)
        })
      })
    })
    describe('notifyTextSize', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyTextSize.options.includes(YooNotify.props.notifyTextSize.default)).toBe(true)
      })
      PropsConfig.notifyTextSize.options.forEach(notifyTextSize => {
        it(`Includes notifyTextSize class: .${classBlock}__text--${notifyTextSize}`, async () => {
          await wrapper.setProps({ notifyTextSize })
          expect(wrapper.find(`.${classBlock}__text--${notifyTextSize}`).exists()).toBe(true)
        })
      })
    })
    describe('showButton', () => {
      it('Has a valid default value', () => {
        expect(YooNotify.props.showButton.default).toBe(false)
      })
      it('Test prop showButton', async () => {
        await wrapper.setProps({ showIndicator: false, showIcon: false, showButton: true })
        expect(wrapper.find('.yoo__container--button').exists()).toBe(true)
      })
    })
  }) // describe Props
})
