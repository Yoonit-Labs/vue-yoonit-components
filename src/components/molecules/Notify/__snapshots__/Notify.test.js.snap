// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooNotify Component Matches Snapshot 1`] = `
<div class="yoo-grid-layout yoo-notify p__t--xs p__b--xs p__r--s p__l--s yoo-grid-layout--h-space-between yoo-grid-layout--v-center yoo-notify--light" style="grid-template-columns: auto 1fr; grid-template-rows: auto; width: 100%; height: auto;">
  <div class="yoo-flex-layout" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: center; align-content: stretch; grid-row: 1; grid-column: 1;" col="1" row="1">
    <div class="yoo-indicator yoo-indicator--normal yoo-indicator--danger yoo-indicator--pulse"></div>
    <!---->
  </div>
  <div class="yoo-flex-layout" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: center; align-content: stretch; grid-row: 1; grid-column: 2;" col="2" row="1">
    <p alignitems="center" class="yoo-notify__text m__r--l m__l--s yoo-notify__text--dark yoo-notify__text--md">Yoo-Notify</p>
  </div>
  <div class="yoo-flex-layout" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: center; align-content: stretch; grid-row: 1; grid-column: 3;" col="3" row="1">
    <div class="yoo__container--button" style="display: none;"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--clear yoo-btn--normal yoo-btn--animation-none"><i class="yoo-btn__icon fas fa-times yoo-btn__icon--normal"></i>
        <p class="yoo-btn__text"></p>
        <!---->
      </button></div>
  </div>
</div>
`;
